src/preferences/upgrade.cpp

```diff
@@ -33,78 +35,107 @@ Upgrade::~Upgrade() {
 
 namespace {
 // mapping to proactively move users to the new all-shader waveform types
-void upgradeToAllShaders(int* waveformType,
-        WaveformWidgetBackend* waveformBackend,
-        allshader::WaveformRendererSignalBase::Options* waveformOptions) {
-    if (*waveformBackend == WaveformWidgetBackend::AllShader) {
-        return;
-    }
-    *waveformOptions = allshader::WaveformRendererSignalBase::None;
-    *waveformBackend = WaveformWidgetBackend::AllShader;
+std::tuple<WaveformWidgetType::Type,
+        WaveformWidgetBackend,
+        allshader::WaveformRendererSignalBase::Options>
+upgradeToAllShaders(int unsafeWaveformType,
+        int unsafeWaveformBackend,
+        int unsafeWaveformOption) {
     // TODO: convert `WaveformWidgetType::Type` to an enum class then shorten more `using enum ...`
     using WWT = WaveformWidgetType;
 
+    if (static_cast<WaveformWidgetBackend>(unsafeWaveformBackend) ==
+            WaveformWidgetBackend::AllShader) {
```

Comment:
technically, this needs to be the other way around (convert an int, not an enum)...

```suggestion
    if (static_cast<int>(WaveformWidgetBackend::AllShader) == unsafeWaveformBackend) {
            
```